src/Umbraco.Web.Common/Extensions/UrlHelperExtensions.cs

```diff
@@ -22,21 +22,27 @@ namespace Umbraco.Extensions;
 
 public static class UrlHelperExtensions
 {
+    /// <summary>
+    /// Gets the Umbraco backoffice URL (if Umbraco is installed).
+    /// </summary>
+    /// <param name="urlHelper">The URL helper.</param>
+    /// <returns>
+    /// The Umbraco backoffice URL.
+    /// </returns>
+    public static string? GetUmbracoBackOfficeUrl(this IUrlHelper urlHelper)
+        => urlHelper.Action("Default", "BackOffice", new { area = Constants.Web.Mvc.BackOfficeArea });
+
     /// <summary>
     ///     Return the back office url if the back office is installed
     /// </summary>
     /// <param name="url"></param>
     /// <returns></returns>
+    /// <remarks>
+    /// This method contained a bug that would result in always returning "/".
+    /// </remarks>
+    [Obsolete("Use the GetUmbracoBackOfficeUrl extension method instead. This method will be removed in Umbraco 13.")]
     public static string? GetBackOfficeUrl(this IUrlHelper url)
-    {
-        var backOfficeControllerType = Type.GetType("Umbraco.Web.BackOffice.Controllers");
-        if (backOfficeControllerType == null)
-        {
-            return "/"; // this would indicate that the installer is installed without the back office
-        }
-
-        return url.Action("Default", ControllerExtensions.GetControllerName(backOfficeControllerType), new { area = Constants.Web.Mvc.BackOfficeApiArea });
```

Comment:
The type `Umbraco.Web.BackOffice.Controllers` doesn't exist in our codebase, but if it did in another assembly, calling `GetControllerName()` on this type would otherwise result in an `InvalidOperationException`:

https://github.com/umbraco/Umbraco-CMS/blob/b69afe81f3f6fcd37480b3b0295a62af44ede245/src/Umbraco.Web.Common/Extensions/ControllerExtensions.cs#L21-L25

Reply:
Reading up on [`Type.GetType()`](https://learn.microsoft.com/en-us/dotnet/api/system.type.gettype?view=net-7.0#system-type-gettype(system-string)), providing a type name qualified by its namespace will only look in the currently executing assembly or in mscorlib.dll/System.Private.CoreLib.dll. This makes the chances of the exception being thrown extremely slim, especially since this method also isn't called in our own codebase.